File: 101/exercises/easy4/century.rb
def cardinal_string(num)
  num_str = num.to_s
  # regular cases
  ending = case num_str[-1]
           when '1' then 'st'
           when '2' then 'nd'
           when '3' then 'rd'
           else 'th'
           end
  # exceptions
  ending = 'th' if %w(11 12 13).include? num_str[-2..-1]
  # result
  num_str + ending
end

def century1(year)
  century, rest = year.divmod 100
  century += 1 if rest > 0
  cardinal_string century
end

def century2(year)
  century = ((year - 1) / 100) + 1
  cardinal_string century
end

# Other alternatives:
# Matt Borgman: century = (year / 100.0).ceil

## Tests

p century1(2000) == '20th'
p century1(2001) == '21st'
p century1(1965) == '20th'
p century1(256) == '3rd'
p century1(5) == '1st'
p century1(10103) == '102nd'
p century1(1052) == '11th'
p century1(1127) == '12th'
p century1(11201) == '113th'

p century2(2000) == '20th'
p century2(2001) == '21st'
p century2(1965) == '20th'
p century2(256) == '3rd'
p century2(5) == '1st'
p century2(10103) == '102nd'
p century2(1052) == '11th'
p century2(1127) == '12th'
p century2(11201) == '113th'

File: 210/small-problems/easy2/century.js

function century(year) {
  var century = String(Math.ceil(year / 100));
  var suffix = 'th';

  if (century.endsWith('1') && !century.endsWith('11')) {
    suffix = 'st';
  } else if (century.endsWith('2')  && !century.endsWith('12')) {
    suffix = 'nd';
  } else if (century.endsWith('3') && !century.endsWith('13')) {
    suffix = 'rd';
  }

  return century + suffix;
}


// Tests

console.log(century(2000) === "20th");
console.log(century(2001) === "21st");
console.log(century(1965) === "20th");
console.log(century(256) === "3rd");
console.log(century(5) === "1st");
console.log(century(10103) === "102nd");
console.log(century(1052) === "11th");
console.log(century(1127) === "12th");
console.log(century(11201) === "113th");
